react/src/utils/Fetch.js

```diff
@@ -0,0 +1,717 @@
+// -------------------------------------------------------------------------------------------------
```

Comment:
Did we write this or where did it come from? It looks awfully detailed to have been just tossed together, so I'm guessing something writes this for us. If that's true, should it be marked as such? (Can it be? Maybe that's part of the price of having something else generate it...